extensions/ql-vscode/src/contextual/queryResolver.ts

```diff
@@ -21,28 +22,87 @@ export async function qlpackOfDatabase(cli: CodeQLCliServer, db: DatabaseItem):
   return await helpers.getQlPackForDbscheme(cli, dbscheme);
 }
 
+/**
+ * Finds the contextual queries with the specified key in a list of CodeQL packs.
+ *
+ * @param cli The CLI instance to use.
+ * @param qlpacks The list of packs to search.
+ * @param keyType The contextual query key of the query to search for.
+ * @returns The found queries from the first pack in which any matching queries were found.
+ */
+async function resolveQueriesFromPacks(cli: CodeQLCliServer, qlpacks: string[], keyType: KeyType): Promise<string[]> {
+  for (const qlpack of qlpacks) {
+    const suiteFile = (await tmp.file({
+      postfix: '.qls'
+    })).path;
+    const suiteYaml = {
+      qlpack,
+      include: {
+        kind: kindOfKeyType(keyType),
+        'tags contain': tagOfKeyType(keyType)
+      }
+    };
+    await fs.writeFile(suiteFile, yaml.safeDump(suiteYaml), 'utf8');
 
-export async function resolveQueries(cli: CodeQLCliServer, qlpack: string, keyType: KeyType): Promise<string[]> {
-  const suiteFile = (await tmp.file({
-    postfix: '.qls'
-  })).path;
-  const suiteYaml = {
-    qlpack,
-    include: {
-      kind: kindOfKeyType(keyType),
-      'tags contain': tagOfKeyType(keyType)
+    const queries = await cli.resolveQueriesInSuite(suiteFile, helpers.getOnDiskWorkspaceFolders());
```

Comment:
We could use the CodeQL CLI server (`codeql execute cli-server`) . Probably not in this pull request, but it could be something to consider in a future refactoring.

Reply:
Actually, did a little bit of digging here and we _are_ using the cli server for all calls to the cli. The queries are handled by the query server and other commands by the cli server.

This means that it is less important to combine these pack references into a single suite.

Reply:
Once the packs are all properly refactored, we'll always find the queries in the dbscheme library pack, so we'll never search the query pack. The only situation where we'll search both packs is with a >=2.6.1 CLI and packs that were released at the same time as 2.6.1. By the next dist upgrade, the contextual queries will already be in the library pack.